compiler/rustc_typeck/src/coherence/inherent_impls.rs

```diff
@@ -60,6 +60,17 @@ impl ItemLikeVisitor<'v> for InherentCollect<'tcx> {
             ty::Dynamic(ref data, ..) if data.principal_def_id().is_some() => {
                 self.check_def_id(item, data.principal_def_id().unwrap());
             }
+            ty::Dynamic(..) => {
+                struct_span_err!(
+                    self.tcx.sess,
+                    ty.span,
+                    E0785,
+                    "cannot define inherent `impl` for a dyn auto trait"
+                )
+                .span_label(ty.span, "impl requires a principal trait")
```

Comment:
```suggestion
                .span_label(ty.span, "impl requires at least one non-auto trait")
```
"Principal trait" is an internal terminology that nobody is familiar with.